rfc/rfc-66/rfc-66.md

```diff
@@ -0,0 +1,165 @@
+Finally, the log name is with the following pattern:
+
+```shell
+${uuid}_${instant}.log.${version}_${task_token}
+```
+
+### The Sorting Rules for Log Files from Different Writers
+
+The sorting rules is important because it somehow decides the natual order processing,
+especially when the event time field are the same, and we don't know which payload to choose when combining.
+Here we can keep using the log file name compactor of current codebase, that is:
+
+```sql
+order by base_commit_time, version_number, write_token
+```
+
+For a single writer, its log files natual order is preserved by the auto increasing version_number,
+for multiple writers, we try the best to preserve the natual order still with the version number,
+but the write_token has deterministic priority in sorting, which breaks the file generation order.
+
+![log file sequence](log_file_sequence.png)
+
+### Format changes
+
+| Type                       | Changes                                                                                                                            |
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| Commit/compaction metadata | No changes                                                                                                                         |
+| Commit file name           | we are only going to add completion time in completed metadata filename for all actions                                            |
+| Log file format            | Adding deltacommit instant time to log file name. The file name does not contain base instant time; No changes to log block format |
+
+### Commit Protocol
+
+| Type                            | Content                                                                                                                                       |
+|---------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| Writer expectations             | Writer does not need to look up the latest file slice to fetch the base commit time anymore, it always uses the current instant time directly |
+| Conflict Resolution             | No explicit conflict resolution, it is handled by the compactor                                                                               |
+| Reader expectations             | Readers still query based on file slices, see the pseudocode for new file slicing                                                             |
+| Synchronization and correctness | Still need some mutex on commit log (timeline). Getting the completion time and adding to the timeline needs to happen atomically             |
+
+## Appendix
+
+### Pseudocode for New File Slicing
+
+```python
```

Comment:
thanks for adding the new file slicing.
let's also add how we handle snapshot, time travel, incremental and cdc queries with the new slicing.